src/utils/hmacValidator.ts

```diff
@@ -37,7 +37,14 @@ class HmacValidator {
         if (notificationRequestItem.additionalData?.[ApiConstants.HMAC_SIGNATURE]) {
             const expectedSign = this.calculateHmac(notificationRequestItem, key);
             const merchantSign = notificationRequestItem.additionalData?.[ApiConstants.HMAC_SIGNATURE];
-            return expectedSign === merchantSign;
+            if(merchantSign && expectedSign.length === merchantSign.length) {
```

Comment:
You can also use Optional Chaining here, for example: 
if(merchantSign?.length === expectedSign.length)